### PR TITLE
generate_text method and pretrained_model_factory

### PR DESCRIPTION
**Pull request by jackattackyang**
# Pull Request

## Description
- Implemented `generate_text` method
- Generated pretrained model weights in `data`
- `pretrained_model_factory` contains different methods for different models and datasets; it calls `text_generator` class.

UPDATE:
- Current model uses 100,000 first characters in each corpus, with 500 embeddings, 150 LSTM layers and 50 epochs.
- `generate_text` can take 2 input parameters: `seed_text` and `pred_len`
- If either is left empty, the model functions as before with random seed and default pred_len of 50
- In the event seed_text is not in the available corpus, seed_text also uses random selection (interestingly LSTM does not care about this, only adjusted for Markov).

#### Define pre-trained factory
`pretrained_factory = PretrainedModelFactory()`

#### Choose model
```
pretrained_model = pretrained_factory.create_pretrained_LSTM_trump_tweets_generator()
pretrained_model = pretrained_factory.create_pretrained_LSTM_shakespeare_text_generator()
pretrained_model = pretrained_factory.create_pretrained_markov_chain_trump_tweets_generator()
pretrained_model = pretrained_factory.create_pretrained_markov_chain_shakespeare_text_generator()
```

#### Generate text with `seed_text` and `pred_len` inputs
`gen_text = pretrained_model.generate_text(seed_text="America", pred_len=50)`
`print(gen_text)`

## Ticket(s)

Affects #39, #44, #45

### CODE REVIEW

File: mimic/text_generator.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 """Core text generator module."""
-from mimic.model.model import Model
+# from mimic.model.model import Model
```
Comment (rwblickhan) [inline mimic/text_generator.py]: If we don't need these we can just get rid of them.

File: mimic/pretrained_model_factory.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+class PretrainedModelFactory:
+    """
+    Pretrained Model Class.
+
+    Class that creates a text generator of a pretrained model chosen by user.
+    """
+
+    def __init__(self):
+        """Initialize a TextGenerator with the given pretrained model type."""
+        logging.basicConfig(format='%(asctime)s %(message)s',
```
Comment (rwblickhan) [inline mimic/pretrained_model_factory.py]: I don't think we need to set the logging config here again.

Comment (jackattackyang) [reply mimic/pretrained_model_factory.py]: removed

File: mimic/text_generator.py (hunk 1 above)
Comment (jackattackyang) [reply mimic/text_generator.py]: Turns out we don't need it because Python is dynamically typed (courtesy of @Annnnnieee). Left it in for clarity.

**Comment by rwblickhan** [comment]
Just out of curiosity, what's the `h5` file format the cleaned data is in?

File: mimic/model/lstm_model.py [hunk 3]
```diff
@@ -32,15 +32,16 @@ def learn(self, text):
         """Use input text to train the LSTM model."""
         # Clean & verify text
         logging.info('Cleaning and verifying text')
-
+        text = text[:150000]
```
Comment (rwblickhan) [inline mimic/model/lstm_model.py]: Not a huge deal, but it would be nice to have a comment here explaining why `150000`/what that choice means in terms of the learning process.

File: mimic/model/lstm_model.py [hunk 4]
```diff
@@ -32,15 +32,16 @@ def learn(self, text):
...
         clean_txt = utils.clean_text(text)
         txt_len = len(clean_txt)
-        utils.verify_text(clean_txt)
+        # utils.verify_text(clean_txt)
```
Comment (rwblickhan) [inline mimic/model/lstm_model.py]: Why was this commented out?

File: mimic/model/lstm_model.py [hunk 5]
```diff
@@ -98,13 +102,33 @@ def predict(self):
...
         or false if an error occured.
         """
+        text = text[:150000]
+        clean_txt = utils.clean_text(text)
+        txt_len = len(clean_txt)
+        # utils.verify_text(clean_txt)
+        self.cleaned_input_text = clean_txt
+
+        corpus = list(clean_txt[0+i:self.seqLen+i] for i in range(
+                                                      0, txt_len,
```
Comment (rwblickhan) [inline mimic/model/lstm_model.py]: Nit: code formatting looks a little weird here.

File: mimic/model/lstm_model.py [hunk 6]
```diff
@@ -98,13 +102,33 @@ def predict(self):
...
+                                                      self.seqLen))
+        self.tokenizer.fit_on_texts(corpus)
+        total_words = len(self.tokenizer.word_index) + 1
+        input_sequences = []
+        for line in corpus:
+            token_list = self.tokenizer.texts_to_sequences(
+                                                             [line])[0]
```
Comment (rwblickhan) [inline mimic/model/lstm_model.py]: Nit: code formatting is a bit off here too.

File: mimic/pretrained_model_factory.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        logging.info("Creating pretrained Markov Chain Shakespeare text"
+                     "generator")
+
+        self.model = TextGenerator(MarkovChainModel())
+        self.model.load_pretrained_model(os.path.join(
+                                 self.pretrained_models_path,
+                                 "trained_markov_model_shakespeare.pickle"))
+        return self.model
+
+    # def generate_LSTM_inclass_variables(self, text):
```
Comment (rwblickhan) [inline mimic/pretrained_model_factory.py]: Is this not needed anymore? If not, I'd rather remove it entirely - no need to keep commented-out stuff in the official repo :)

File: mimic/util.py [hunk 8]
```diff
@@ -20,4 +20,5 @@ def verify_text(clean_text):
     """Verify that the text is suitable for ML."""
     if len(clean_text.split()) < 20000:
         # TODO should we make a custom error class?
+        print(len(clean_text.split()))
```
Comment (rwblickhan) [inline mimic/util.py]: This should probably be a call to `logging`, not `print`.

File: mimic/model/lstm_model.py (hunk 4 above)
Comment (jackattackyang) [reply mimic/model/lstm_model.py]: This was checking word count to be > 20000 which wasn't satified for smaller corpora, but we can add it back in now

**Comment by jackattackyang** [comment]
`h5` is the default HDF5 file format supported by Keras. Good for high dimensional array data.

File: mimic/pretrained_model_factory.py (hunk 7 above)
Comment (jackattackyang) [reply mimic/pretrained_model_factory.py]: removed

File: mimic/util.py (hunk 8 above)
Comment (jackattackyang) [reply mimic/util.py]: Thanks for the catch; that was just for testing
